### PR TITLE
chore: Refresh button for Topics and Consumer Groups UI

### PR DESCRIPTION
**Pull request by Kells512**
**What this PR does / why we need it**:

> A refresh button that can be used for Consumer Groups and Topics

**Verification steps** 

> **Hoover**
> < PROOFOFCONCEPTS > < RefreshButton > < Example1 >
>Screenshot from 2022-09-19 13-35-54

> **Text next to button**
> < PROOFOFCONCEPTS > < RefreshButton > < Example2 >
> Screenshot from 2022-09-19 13-36-10

> **Text to far right*
> < PROOFOFCONCEPTS > < RefreshButton > < Example3  >
> Screenshot from 2022-09-19 13-36-20

### CODE REVIEW

**Comment by Kells512** [comment]
@jgiardino I had some issues with the last pr so I had to make a new one

File: src/ProofOfConcepts/RefreshButton/RefreshButton.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        {isRefreshing ? (
+          <span className="pf-c-button__progress spinner_position">
+            <Spinner size="md" />
+          </span>
+        ) : (
+          <span className="pf-c-button__progress spinner_position">
+            <SyncAltIcon />
+          </span>
+        )}
+      </Button>
```
[the remark this answers is not in the document]
Comment (mcoker) [reply src/ProofOfConcepts/RefreshButton/RefreshButton.tsx:40]: As long as you're on `@patternfly/react-core: 4.198.5` or greater, the `<Button>` component supports the `isLoading` prop, which should handle all of this for you. The only catch with the `plain` variant button is that the icon either needs to be passed via the `icon` prop (`children` needs to be `null`), or `isLoading` needs to be `null` (not `false`) when `!isRefreshing`.

I'm curious if this change will work for you? Or was there another reason you needed to handle this differently?

Also I opened this issue upstream to allow you to do `<Button isLoading={isRefreshing}><SyncAltIcon /></Button>` - https://github.com/patternfly/patternfly-react/issues/8228

**Comment by riccardo-forina** [comment]
Just chiming in to say that we already have a [Refresh button](https://redhat-developer.github.io/app-services-ui-components/?path=/story/shared-refreshbutton--default). I think it would be helpful for the PoC to be placed in the PoC folder only if they are things really new or experimental. Improvements over existing elements should be done on the existing code.

File: src/ProofOfConcepts/RefreshButton/RefreshButton.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kells512) [reply src/ProofOfConcepts/RefreshButton/RefreshButton.tsx]: I applied those comments and they seem to be okay

**Comment by jgiardino** [comment]
Thanks for making updates, @Kells512! And thanks to @mcoker for reviewing 🤗  

> Just chiming in to say that we already have a [Refresh button](https://redhat-developer.github.io/app-services-ui-components/?path=/story/shared-refreshbutton--default). I think it would be helpful for the PoC to be placed in the PoC folder only if they are things really new or experimental. Improvements over existing elements should be done on the existing code.

Just wanted to check if any action was taken on this comment? I'm not sure how to actually implement @riccardo-forina's suggestion. I think of POC as a place for including preliminary components that might not be production ready yet, and I'm not sure if that's the case with this PR. 🤷‍♀️  I also think that @Kells512's updates would likely want to use the existing refresh button (but I didn't dig into the code to see if that's the case).

**Comment by Kells512** [comment]
> Thanks for making updates, @Kells512! And thanks to @mcoker for reviewing hugs
> 
> > Just chiming in to say that we already have a [Refresh button](https://redhat-developer.github.io/app-services-ui-components/?path=/story/shared-refreshbutton--default). I think it would be helpful for the PoC to be placed in the PoC folder only if they are things really new or experimental. Improvements over existing elements should be done on the existing code.
> 
> Just wanted to check if any action was taken on this comment? I'm not sure how to actually implement @riccardo-forina's suggestion. I think of POC as a place for including preliminary components that might not be production ready yet, and I'm not sure if that's the case with this PR. woman_shrugging I also think that @Kells512's updates would likely want to use the existing refresh button (but I didn't dig into the code to see if that's the case).

I updated the Refreshbutton.tsx in the shared folder to include my changes for the examples. The examples threw a lot of errors when moved to the same folder, so I thought it could be easier to keep them in the POC. Since they are just some examples made to show different styles for the updates.

**Comment by Kells512** [comment]
Hey @riccardo-forina, would you mind giving a look at the code after the updates.

**Review by riccardo-forina — APPROVED** [review]
LGTM, but I left a comment about a change in the `.npmrc` file that I'd like to know more about (and probably see removed)

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+legacy-peer-deps=true
```
[the remark this answers is not in the document]
Comment (Kells512) [reply .npmrc:1]: I'm not sure why that appeared, I will remove that line.

**Pull request closed without merging**